analyzers/src/SonarAnalyzer.CSharp/Rules/DisposableNotDisposed.cs

```diff
@@ -158,10 +158,12 @@ private static bool IsNodeInsideUsingStatement(SyntaxNode node)
         {
             var ancestors = node.AncestorsAndSelf().ToArray();
             var usingStatements = ancestors.OfType<UsingStatementSyntax>();
-            var usingDeclarations = ancestors.OfType<LocalDeclarationStatementSyntax>();
+            var usingDeclarations = ancestors
+                .OfType<LocalDeclarationStatementSyntax>()
+                .Where(x => x.UsingKeyword() != default);
 
             return usingStatements.Any(x => ancestors.Contains(x.Expression) || ancestors.Contains(x.Declaration))
-                || usingDeclarations.Any(x => ancestors.Contains(x.Declaration));
+                || usingDeclarations.Any();
```

Comment:
let's add more noncompliant test cases with combinations like

```
            await using ((fs5 = new FileStream(@"c:\foo.txt", FileMode.Open)).ConfigureAwait(false))
            {

                FileStream bam = new FileStream(@"c:\foo.txt", FileMode.Open); // Noncompliant
                using (...)
                {
                  bam = new FileStream(@"c:\foo.txt", FileMode.Open); // Noncompliant
                 }
            }

```

Reply:
Added it to the test cases.